internal/proxy/proxy.go

```diff
@@ -187,38 +199,89 @@ func (c *Config) DialOptions(i InstanceConnConfig) []cloudsqlconn.DialOption {
 	return opts
 }
 
-// DialerOptions builds appropriate list of options from the Config
-// values for use by cloudsqlconn.NewClient()
-func (c *Config) DialerOptions(l cloudsql.Logger) ([]cloudsqlconn.Option, error) {
-	opts := []cloudsqlconn.Option{
-		cloudsqlconn.WithUserAgent(c.UserAgent),
+func (c *Config) credentialsOpt(l cloudsql.Logger) (cloudsqlconn.Option, error) {
```

Comment:
This doesn't look like it modifies c in anyway. Would it be better to take c as an arg instead?

Reply:
Yes, I'm in favor of that idea.

Reply:
Looking at this again, I see we're reading a bunch of data from `c`. So even though we're not modifying it, it's still common to define `c` as a receiver. Passing it as an argument would basically be a de-sugared version of the same code here.

Reply:
downgrade to nit: but it does seem like a receiver indicates we are doing something to `c`, where a read only parameter makes clear in the intent is only to create something from c.

Reply:
Let me try that in a separate PR and we can apply the pattern across the board if we like it.

Reply:
https://github.com/GoogleCloudPlatform/cloud-sql-proxy/pull/1480